flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -1599,6 +1603,33 @@ private OptionalLong restoreLatestCheckpointedStateInternal(
         }
     }
 
+    private Map<OperatorID, OperatorState> extractOperatorStates(CompletedCheckpoint checkpoint) {
+        Map<OperatorID, OperatorState> operatorStates = checkpoint.getOperatorStates();
+
+        if (checkpoint.getCheckpointID() == checkpointIdOfIgnoredInFlightData) {
```

Comment:
If not configured, this field is initialized to 0, right?

And here we use a simple equality check against checkpoint ID. This is not a problem currently, as all checkpoint counters start from 1. But it's an implementation detail (of counters) and I think it's too risky to rely on it.

So I'd initialize it to -1 (maybe in configs).

Reply:
Yes, it makes sense